src/vendorpromo/models.py

```diff
@@ -112,6 +113,25 @@ def get_display_code(self):
     def redemptions(self):
         return self.invoice.filter(status=InvoiceStatus.COMPLETE)
 
+    def does_offer_apply(self, offer):
+        if self.promo.applies_to.products.filter(pk__in=offer.products.all().values_list('pk', flat=True)).count():
```

Comment:
shouldn't need the count(), i think- a non empty queryset should evaluate to true